### PR TITLE
refactor: sort active proposals by DAO and expiration

### PR DESCRIPTION
**Pull request by Zied-Dahmani**
## Scope

[Ticket](https://github.com/hypha-dao/hypha-wallet/issues/300)

I have implemented the logic to `sort` active proposals by `DAO name` and `expiration` date.

### CODE REVIEW

File: lib/core/extension/proposal_model_extensions.dart [hunk 1]
```diff
@@ -6,8 +6,12 @@ extension ProposalModelTimeFormatting on ProposalModel {
 
     final now = DateTime.now();
     final expirationDate = expiration!.toLocal();
-    final difference = expirationDate.difference(now);
 
+    if (expirationDate.isBefore(now)) {
+      return 'Expired';
```
Comment (n13) [inline lib/core/extension/proposal_model_extensions.dart:11]: Not sure this is the best term - is this UX visible?

It should be the same as on the web... maybe passed of failed?

File: lib/core/network/repository/proposal_repository.dart [hunk 2]
```diff
@@ -10,35 +11,72 @@ class ProposalRepository {
...
+      final daoNameComparison = a.daoName.compareTo(b.daoName);
+      if (daoNameComparison != 0) {
+        return daoNameComparison;
+      }
+
+      final now = DateTime.now();
+
+      final isAExpired = a.expiration != null && a.expiration!.isBefore(now);
+      final isBExpired = b.expiration != null && b.expiration!.isBefore(now);
+
```
Comment (n13) [inline lib/core/network/repository/proposal_repository.dart:72]: if we have the concept that a proposal can be expired then that should be handled in a utility function so we don't repeat this code everywhere (isBefore(now))

Comment (n13) [reply lib/core/network/repository/proposal_repository.dart:72]: I recommend an extension on ProposalModel "isExpired"

**Review by n13 — CHANGES_REQUESTED** [review]
refactor so isExpired code is only in 1 place (see comments)

**Comment by Zied-Dahmani** [comment]
> refactor so isExpired code is only in 1 place (see comments)

You are absolutely right about the comments. However, I’m marking this as a 'to-do' item (which is why I haven’t implemented it as an extension yet). I believe that active proposals should not include expired ones.

**Review by n13 — APPROVED** [review]
Ok - refactor the isExpired fucntion is a To-Do item

**Review by n13 — APPROVED** [review]
I see changes were made. 

LGTM!

**Pull request merged**